internal/antewrapper/min_gas_prices_decorator.go

```diff
@@ -0,0 +1,74 @@
+package antewrapper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MinGasPricesDecorator will check if the transaction's fee is at least as large
+// as the local validator's minimum gasFee (defined in validator config).
+// If fee is too low, decorator returns error and tx is rejected from mempool.
+// Note this only applies when ctx.CheckTx = true
+// If fee is high enough or not CheckTx, then call next AnteHandler
+// CONTRACT: Tx must implement FeeTx to use MinGasPricesDecorator
+type MinGasPricesDecorator struct{}
+
+func NewMinGasPricesDecorator() MinGasPricesDecorator {
+	return MinGasPricesDecorator{}
+}
+
+func (mfd MinGasPricesDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !simulate {
+		err := checkTxFeeWithValidatorMinGasPrices(ctx, tx)
+		if err != nil {
+			return ctx, err
+		}
+	}
+
+	return next(ctx, tx, simulate)
+}
+
+// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set by each validator, can the tx priority is computed from the gas price.
```

Comment:
Yup.... Copied that directly from the SDK.
But since I changed the signature of the function and also removed the priority part, it makes sense to update that comment too.
Fixed.